src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClient.cs

```diff
@@ -498,5 +500,65 @@ private static void ValidateRoleDefinition(PSRoleDefinition roleDefinition)
                 throw new ArgumentException(ProjectResources.InvalidActions);
             }
         }
+        public static void ValidateScope(string scope)
```

Comment:
yes it will be better if service side can return the correct error message. however this is not going to happen in a short time.

here we want do some very basic validations which is not going to be changed by service side for a while so end user will not get confused (for most of below cases, ARM the service side only returns an error message like "unsupported API version".

we are involving ARM folks to review this as well.

---
In reply to: [119177756](https://github.com/Azure/azure-powershell/pull/4019#discussion_r119177756) [](ancestors = 119177756)